File: compiler/src/main/java/com/aaroncoplan/waterfall/compiler/CompilerError.java
package com.aaroncoplan.waterfall.compiler;

/**
 * Thrown by {@link Main#run(String[])} when compilation fails for any reason
 * (bad arguments, missing files, syntax errors, verification errors).
 *
 * The wrapper {@link Main#main(String[])} catches this and exits non-zero so
 * the {@code ./waterfall} script signals failure to its caller. Tests that
 * invoke the compiler in-process call {@code Main.run(...)} directly so they
 * can observe (or assert against) the failure without ending the JVM.
 */
public class CompilerError extends RuntimeException {

    private static final long serialVersionUID = 1L;

    public CompilerError(String message) {
        super(message);
    }
}

File: compiler/src/main/kotlin/com/aaroncoplan/waterfall/compiler/CompilerError.kt
package com.aaroncoplan.waterfall.compiler

/**
 * Thrown by [Main.run] when compilation fails for any reason
 * (bad arguments, missing files, syntax errors, verification errors).
 *
 * The wrapper [Main.main] catches this and exits non-zero so the
 * `./waterfall` script signals failure to its caller. Tests that invoke the
 * compiler in-process call [Main.run] directly so they can observe (or assert
 * against) the failure without ending the JVM.
 */
class CompilerError(message: String) : RuntimeException(message)
